src/main/java/duke/data/exception/DukeException.java

```diff
@@ -0,0 +1,11 @@
+package duke.data.exception;
+
+/**
+ * Represents an exception to be thrown when invalid inputs are provided by the user
+ * or when there is an error loading or saving the list of tasks.
+ */
+public class DukeException extends Exception {
```

Comment:
Hiii, I think it is good to use DukeException to handle the exception in your code. And there is a small suggestion that maybe you could use some child-classes of DukeException to handle different exceptions, like TaskNotExistException. In this way, it could be more well-organized : )